src/main/scala/beam/sim/BeamHelper.scala

```diff
@@ -706,6 +699,33 @@ trait BeamHelper extends LazyLogging {
       PopulationScaling.samplePopulation(scenario, beamScenario, beamServices.beamConfig, beamServices, outputDir)
     }
 
+    if (beamScenario.beamConfig.beam.agentsim.snapLocationAndRemoveInvalidInputs) {
+      logger.info(s"""
+      |The parameter `beam.agentsim.snapLocationAndRemoveInvalidInputs` is enabled.
+      |This may take some time to finish depending on the size of population/households.""".stripMargin)
+
+      val beforeHouseholdsCount = scenario.getHouseholds.getHouseholds.size()
+      val beforePopulationCount = scenario.getPopulation.getPersons.size()
+
+      val snapLocationHelper = SnapLocationHelper(
+        new GeoUtilsImpl(beamScenario.beamConfig),
+        beamScenario.transportNetwork.streetLayer,
+        beamScenario.beamConfig.beam.routing.r5.linkRadiusMeters
+      )
+
+      ScenarioLoaderHelper.validateScenario(scenario, snapLocationHelper, Some(outputDir))
+
+      val afterHouseholdsCount = scenario.getHouseholds.getHouseholds.size()
+      val afterPopulationCount = scenario.getPopulation.getPersons.size()
+
+      logger.info(
+        s"""
+      |After snapping locations and validating scenario:
+      |Number of households: $afterHouseholdsCount. Removed: ${beforeHouseholdsCount - afterHouseholdsCount}.
+      |Number of persons: $afterPopulationCount. Removed: ${beforePopulationCount - afterPopulationCount}.""".stripMargin
+      )
+    }
+
     // write static metrics, such as population size, vehicles fleet size, etc.
     // necessary to be called after population sampling
     BeamStaticMetricsWriter.writeSimulationParameters(
```

Comment:
FYI, I removed the same method call within `samplePopulation` because it was already here.